### PR TITLE
CCAP-57 Add new referral service screen

### PR DESCRIPTION
**Pull request by coltborg**
#### 🔗 Jira ticket
[CCAP-57](https://codeforamerica.atlassian.net/browse/CCAP-57)

#### ✍️ Description
NOTE: this ticket will be merged into #580, as it's related to appendix page work.

- Add new referral service screen
- Add new appendix page fields
- Update tests

#### 📷 Design reference
[Figma](https://www.figma.com/design/1oRFY7p6tdGogBJk7Ugle8/IL-CCAP-Families-Full-Flow?node-id=4201-16751)

#### ✅ Completion tasks

- [x] Added relevant tests
- [x] Meets acceptance criteria

[CCAP-57]: https://codeforamerica.atlassian.net/browse/CCAP-57?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/main/java/org/ilgcc/app/pdf/ApplicationPreparer.java [hunk 1]
```diff
@@ -3,6 +3,7 @@
 import static org.ilgcc.app.utils.SubmissionUtilities.formatToStringFromLocalDate;
 
 import formflow.library.data.Submission;
+import formflow.library.pdf.CheckboxField;
```
Comment (analoo) [inline src/main/java/org/ilgcc/app/pdf/ApplicationPreparer.java:6]: @coltborg - doesn't look like you are using this field.

```suggestion
```

File: src/main/java/org/ilgcc/app/pdf/ApplicationPreparer.java [hunk 2]
```diff
@@ -76,6 +78,20 @@ public Map<String, SubmissionField> prepareSubmissionFields(Submission submissio
                 results.put("unearnedIncomePrograms-housing-vouchers", new SingleField("unearnedIncomePrograms-housing-vouchers", "true", null));
             }
         }
+
+        List<String> unearnedIncomeReferral = (List<String>) inputData.getOrDefault("unearnedIncomeReferralServices[]", List.of());
+        if (!unearnedIncomeReferral .isEmpty()){
```
Comment (analoo) [inline src/main/java/org/ilgcc/app/pdf/ApplicationPreparer.java:83]: up to you, but it doesn't seem like this is necessary. If the list is empty, all of the subsequent if statements will return false.

File: src/main/java/org/ilgcc/app/pdf/ApplicationPreparer.java (hunk 1 above)
Comment (coltborg) [reply src/main/java/org/ilgcc/app/pdf/ApplicationPreparer.java:6]: Will remove 👍🏻

File: src/main/resources/templates/gcc/unearned-income-referral-services.html [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+        subtext=#{unearned-income-referral-services.subtext},
+        formAction=${formAction},
+        buttonLabel=#{general.button.continue},
+        inputContent=~{::inputContent})}">
+            <th:block th:ref="inputContent">
+                <th:block th:replace="~{fragments/inputs/checkboxFieldset ::
+                      checkboxFieldset(inputName='unearnedIncomeReferralServices',
+                        label='',
+                        ariaLabel='header',
+                        options=${referralServices},
```
Comment (analoo) [inline src/main/resources/templates/gcc/unearned-income-referral-services.html:15]: I love how elegant the screen with one input is and that the checkboxFieldset will take in all the options as an input.

File: src/main/java/org/ilgcc/app/pdf/ApplicationPreparer.java (hunk 2 above)
Comment (coltborg) [reply src/main/java/org/ilgcc/app/pdf/ApplicationPreparer.java:83]: That's true, I'll remove it.

**Review by analoo — APPROVED** [review]

**Pull request merged**